README.md

```diff
@@ -27,165 +27,184 @@ curl http://my-url
 
 ```sh
 cd ..
 ```
 
-and you can deploy it from your local machine using the `gcloud` command-line tool.
+and you can deploy it from your local machine using the `gcloud` command-line
+tool.
 [Check out the Cloud Functions quickstart](https://cloud.google.com/functions/docs/quickstart).
 
-## Container environments based on Knative
+### Container environments based on Knative
 
 The Functions Framework is designed to be compatible with Knative environments.
-Just build and deploy your container to a Knative environment. Note that your app needs to listen
-`PORT` environment variable per [Knative runtime contract](https://github.com/knative/serving/blob/master/docs/runtime-contract.md#inbound-network-connectivity).
+Just build and deploy your container to a Knative environment. Note that your
+app needs to listen `PORT` environment variable per the
+[Knative runtime contract](https://github.com/knative/serving/blob/master/docs/runtime-contract.md#inbound-network-connectivity).
 
-# Configure the Functions Framework
+## Configure the Functions Framework
 
-If you're deploying to Google Cloud Functions, you don't need to worry about writing a
-`package main`. But if you want to run your function locally (e.g., for local development),
-you may want to configure the port, the function to be executed, and the function signature type
-(which specifies event unmarshalling logic). You can do this by modifying the `main.go`
-file described above:
+If you're deploying to Google Cloud Functions, you don't need to worry about
+writing a `package main`. But if you want to run your function locally (e.g.,
+for local development), you may want to configure the port, the function to be
+executed, and the function signature type (which specifies event unmarshalling
+logic). You can do this by modifying the `main.go` file described above:
 
 To select a port, set the `$PORT` environment variable when running.
 
 ```sh
 PORT=8000 ./cmd
 ```
 
-To select a function, pass your function to `funcframework.RegisterHTTPFunction` in the second variable.
+To select a function, pass your function to
+`funcframework.RegisterHTTPFunctionContext` in the second variable.
 
 ```golang
-funcframework.RegisterHTTPFunction("/", myFunction);
+if err := funcframework.RegisterHTTPFunctionContext("/", myFunction); err != nil {
+  log.Fatalf(err)
+}
 ```
 
-If your function handles events, use `funcframework.RegisterEventFunction` instead of `funcframework.RegisterHTTPFunction`.
+If your function handles events, use
+`funcframework.RegisterEventFunctionContext` instead of
+`funcframework.RegisterHTTPFunctionContext`.
 
 ```golang
-funcframework.RegisterEventFunction("/", eventFunction);
+if err := funcframework.RegisterEventFunctionContext("/", eventFunction); err != nil {
+  log.Fatalf(err)
+}
 
-func eventFunction(ctx context.Context, e myEventType){
-	// function logic
+func eventFunction(ctx context.Context, e myEventType) error {
+    // function logic
```

Comment:
Indenting is not consistent with the samples in this PR. I'd expect all samples to have the same indentation width.